2 Класса с решением и 1 класс в качестве точки запуска(Main - main).

ApprehensiveBug opened the pull request:
Поместил класс Main в пакет номер-задачи.
Создал классы - Book, LibrarySystem.
Реализовал логику - ООП.

ApprehensiveBug committed:
Поместил класс Main в пакет номер-задачи.
Создал классы - Book, LibrarySystem.
Реализовал логику - ООП.

MarselYsup commented on src/main/java/school/faang/bjs2_86033/Book.java:
```diff
@@ -0,0 +1,37 @@
+import java.util.Objects;
+
+public record Book(String title, String author, int year) {
+    public Book(String title, String author, int year) {
```
```record``` автоматически генерирует конструктор, equals, hashCode, toString тебе не нужно их переопределять

MarselYsup commented on src/main/java/school/faang/bjs2_86033/LibrarySystem.java:
```diff
@@ -0,0 +1,28 @@
+import java.util.Map;
+
+public class LibrarySystem {
+    private static final HashMap<Book, String> BOOK_LIBRARY = new HashMap<>();
```
если предполагается различные экземпляры класса LibrarySystem , то тебе не нужно делать методы static, в ином случае советую сделать конструктор приватным, так никто не сможет создать различные экземпляры класса и чтобы вызвать методы используй название класса LibrarySystem.addBook("Как закалялась сталь", "В.Островский", 1948, 2);

MarselYsup commented on src/main/java/school/faang/bjs2_86033/LibrarySystem.java:
```diff
@@ -0,0 +1,28 @@
+    private static final HashMap<Book, String> BOOK_LIBRARY = new HashMap<>();
+
+    public static void addBook(String title, String author, int year, int location) {
+        String index = "" + location;
```
логика изменения типа данных локации выглядит сложным, ты можешь просто принимать изначально String location в аргументы метода

MarselYsup commented on src/main/java/school/faang/bjs2_86033/LibrarySystem.java:
```diff
@@ -0,0 +1,28 @@
+
+    public static void printAllBooks() {
+        for (Map.Entry<Book, String> keyValue : BOOK_LIBRARY.entrySet()) {
+            System.out.println(keyValue.getKey() + " [" + keyValue.getValue() + "]");
```
Так как строки неизменяемый тип данных, то при конкатенации ты создаешь лишние строки в памяти, тут можно воспользоваться форматированным выводом (PS можешь про него прочитать )

```java
            System.out.printf("%s [%s]%n",keyValue.getKey(), keyValue.getValue());
```

ApprehensiveBug committed:
Рекордный класс реализован корректно, НО, я решил изменить toString, такой вывод мне больше нравится.

Все мои ошибки, указанные вами, бли исправлены. (Я совсем забыл про форматирование тогда, изменение типа location в методе addBook я убрал и теперь там просто строка принимается, конструктор LibrarySystem теперь приватный - никто не сможет создавать экземпляры, методы вызываются через LibrarySystem, ДОБАВИЛ ПРОВЕРКИ в методы findBook и removeBook - реагируют если искомой/удаляемой книги нет в Библиотеке.

ApprehensiveBug committed:
Рекордный класс реализован корректно, НО, я решил изменить toString, такой вывод мне больше нравится.

Все мои ошибки, указанные вами, бли исправлены. (Я совсем забыл про форматирование тогда, изменение типа location в методе addBook я убрал и теперь там просто строка принимается, конструктор LibrarySystem теперь приватный - никто не сможет создавать экземпляры, методы вызываются через LibrarySystem, ДОБАВИЛ ПРОВЕРКИ в методы findBook и removeBook - реагируют если искомой/удаляемой книги нет в Библиотеке.
